### PR TITLE
Consistently use `suggest_float`

### PR DESCRIPTION
**Pull request by crcrpar**
In the pursuit of the comprehensive use of  `suggest_float` instead of `suggest_( | log | discrete)uniform`.

### CODE REVIEW

**Comment by HideakiImamura** [comment]
@himkt Could you review this PR if you have time? Please feel free to remove the assignment if you are busy.

File: tests/integration_tests/test_skopt.py [hunk 1]
```diff
@@ -27,27 +27,27 @@ def test_conversion_from_distribution_to_dimension() -> None:
...
 
         dimensions = [
-            # Original: trial.suggest_uniform('p0', -3.3, 5.2)
+            # Original: trial.suggest_float('p0', -3.3, 5.2)
             space.Real(-3.3, 5.2),
-            # Original: trial.suggest_uniform('p1', 2.0, 2.0)
+            # Original: trial.suggest_float('p1', 2.0, 2.0)
             # => Skipped because `skopt.Optimizer` cannot handle an empty `Real` dimension.
-            # Original: trial.suggest_discrete_uniform('p9', 2.2, 2.2, 0.5)
+            # Original: trial.suggest_float('p9', 2.2, 2.2, size=0.5)
```
Comment (HideakiImamura) [inline tests/integration_tests/test_skopt.py:34]: A nit
```suggestion
            # Original: trial.suggest_float('p9', 2.2, 2.2, step=0.5)
```

**Review by HideakiImamura — COMMENTED** [review]
Thanks for the code-fix. I have a minor comment. PTAL.

**Review by HideakiImamura — APPROVED** [review]
LGTM.

**Comment by himkt** [comment]
Sorry but I want to clarify what `consistently use` means. 🙏 
Is the scope of this PR is to replace `suggest_{log,discrete_}uniform` (a) with `suggest_float` (b) where (a) and (b) both appear in the same script, right? (I'm asking because I found there were still `suggest_{log,discrete_}uniform` in Optuna)

<details>

```bash
> git grep "suggest_discrete_uniform"
optuna/distributions.py:    This object is instantiated by :func:`~optuna.trial.Trial.suggest_discrete_uniform`, and passed
optuna/integration/chainermn.py:    def suggest_discrete_uniform(self, name: str, low: float, high: float, q: float) -> float:
optuna/integration/chainermn.py:            return self.delegate.suggest_discrete_uniform(name, low, high, q)
optuna/multi_objective/trial.py:    def suggest_discrete_uniform(self, name: str, low: float, high: float, q: float) -> float:
optuna/multi_objective/trial.py:        Please refer to the documentation of :func:`optuna.trial.Trial.suggest_discrete_uniform`
optuna/multi_objective/trial.py:        return self._trial.suggest_discrete_uniform(name, low, high, q)
optuna/trial/_base.py:    def suggest_discrete_uniform(self, name: str, low: float, high: float, q: float) -> float:
optuna/trial/_fixed.py:    def suggest_discrete_uniform(self, name: str, low: float, high: float, q: float) -> float:
optuna/trial/_frozen.py:    def suggest_discrete_uniform(self, name: str, low: float, high: float, q: float) -> float:
optuna/trial/_trial.py:        :func:`~optuna.trial.Trial.suggest_discrete_uniform`.
optuna/trial/_trial.py:            :func:`~optuna.trial.Trial.suggest_discrete_uniform`.
optuna/trial/_trial.py:                    this method falls back to :func:`~optuna.trial.Trial.suggest_discrete_uniform`.
optuna/trial/_trial.py:                return self.suggest_discrete_uniform(name, low, high, step)
optuna/trial/_trial.py:    def suggest_discrete_uniform(self, name: str, low: float, high: float, q: float) -> float:
optuna/trial/_trial.py:                    subsample = trial.suggest_discrete_uniform("subsample", 0.1, 1.0, 0.1)
tests/integration_tests/test_chainermn.py:    def test_suggest_discrete_uniform(storage_mode: str, comm: CommunicatorBase) -> None:
tests/integration_tests/test_chainermn.py:                x1 = mn_trial.suggest_discrete_uniform("x", low, high, q)
tests/integration_tests/test_chainermn.py:                x2 = mn_trial.suggest_discrete_uniform("x", low, high, q)
tests/integration_tests/test_sampler.py:        p3 = trial.suggest_discrete_uniform("p3", 0, 9, 3)
tests/integration_tests/test_sampler.py:        p3 = trial.suggest_discrete_uniform("p3", 0, 9, 3)
tests/integration_tests/test_sampler.py:        p3 = trial.suggest_discrete_uniform("p3", 0, 9, 3)
tests/integration_tests/test_sampler.py:    p6 = trial.suggest_discrete_uniform("p6", 10, 20, 2)
tests/integration_tests/test_sampler.py:    p7 = trial.suggest_discrete_uniform("p7", 0.1, 1.0, 0.1)
tests/integration_tests/test_sampler.py:    p8 = trial.suggest_discrete_uniform("p8", 2.2, 2.2, 0.5)
tests/samplers_tests/test_grid.py:        d = trial.suggest_discrete_uniform("d", -5, 5, 1)
tests/samplers_tests/tpe_tests/test_sampler.py:        t.suggest_discrete_uniform("c", 1.0, 100.0, 3.0)
tests/trial_tests/test_fixed.py:def test_suggest_discrete_uniform() -> None:
tests/trial_tests/test_fixed.py:    assert trial.suggest_discrete_uniform("x", 0.0, 1.0, 0.1) == 0.9
tests/trial_tests/test_fixed.py:        trial.suggest_discrete_uniform("y", 0.0, 1.0, 0.1)
tests/trial_tests/test_frozen.py:        c = trial.suggest_discrete_uniform("c", 0.0, 10.0, 1.0)
tests/trial_tests/test_frozen.py:def test_suggest_discrete_uniform() -> None:
tests/trial_tests/test_frozen.py:    assert trial.suggest_discrete_uniform("x", 0.0, 1.0, 0.1) == 0.9
tests/trial_tests/test_frozen.py:        trial.suggest_discrete_uniform("y", 0.0, 1.0, 0.1)
tests/trial_tests/test_trial.py:    x6 = trial.suggest_discrete_uniform("x3", 10, 20, 1.0)
tests/trial_tests/test_trial.py:def test_check_distribution_suggest_discrete_uniform(
tests/trial_tests/test_trial.py:        trial.suggest_discrete_uniform("x", 10, 20, 2)
tests/trial_tests/test_trial.py:        trial.suggest_discrete_uniform("x", 10, 20, 2)
tests/trial_tests/test_trial.py:        trial.suggest_discrete_uniform("x", 10, 22, 2)
tests/trial_tests/test_trial.py:def test_suggest_discrete_uniform(storage_init_func: Callable[[], storages.BaseStorage]) -> None:
tests/trial_tests/test_trial.py:        assert trial.suggest_discrete_uniform("c", 1.0, 1.0, 1.0) == 1.0  # Suggesting a param.
tests/trial_tests/test_trial.py:            trial.suggest_discrete_uniform("c", 1.0, 1.0, 1.0) == 1.0
tests/trial_tests/test_trial.py:def test_suggest_discrete_uniform_range(
tests/trial_tests/test_trial.py:            x = trial.suggest_discrete_uniform(
tests/trial_tests/test_trial.py:            x = trial.suggest_discrete_uniform(
2021-02-18 23:37:41 [~/work/github.com/himkt/optuna] @ pr
> git grep "trial.suggest_loguniform"
optuna/multi_objective/trial.py:        return self._trial.suggest_loguniform(name, low, high)
optuna/trial/_trial.py:                    c = trial.suggest_loguniform("c", 1e-5, 1e2)
tests/importance_tests/test_mean_decrease_impurity.py:    x2 = trial.suggest_loguniform("x2", 0.1, 3)
tests/importance_tests/test_mean_decrease_impurity.py:    x3 = trial.suggest_loguniform("x3", 2, 4)
tests/integration_tests/test_chainermn.py:        y = trial.suggest_loguniform("y", 20, 30)
tests/integration_tests/test_chainermn.py:                    mn_trial.suggest_loguniform("x1", low1, high1)
tests/integration_tests/test_chainermn.py:                x4 = mn_trial.suggest_loguniform("x2", low2, high2)
tests/integration_tests/test_chainermn.py:                    mn_trial.suggest_loguniform("x", low, high)
tests/integration_tests/test_chainermn.py:                x1 = mn_trial.suggest_loguniform("x", low, high)
tests/integration_tests/test_chainermn.py:                x2 = mn_trial.suggest_loguniform("x", low, high)
tests/integration_tests/test_mlflow.py:    y = trial.suggest_loguniform("y", 20, 30)
tests/integration_tests/test_mlflow.py:    y = trial.suggest_loguniform("y", 20, 30)
tests/integration_tests/test_sampler.py:        p1 = trial.suggest_loguniform("p1", 1, 10)
tests/integration_tests/test_sampler.py:        p1 = trial.suggest_loguniform("p1", 1, 10)
tests/integration_tests/test_sampler.py:        p1 = trial.suggest_loguniform("p1", 50, 100)  # The range has been changed
tests/integration_tests/test_sampler.py:    p2 = trial.suggest_loguniform("p2", 0.0001, 0.3)
tests/integration_tests/test_sampler.py:    p3 = trial.suggest_loguniform("p3", 1.1, 1.1)
tests/integration_tests/test_tensorboard.py:    y = trial.suggest_loguniform("y", 20.0, 30.0)
tests/samplers_tests/test_grid.py:        e = trial.suggest_loguniform("e", 0.0001, 1)
tests/samplers_tests/test_samplers.py:        assert trial.suggest_loguniform("d", 1, 100) == unknown_param_value
tests/study_tests/test_study.py:    y = trial.suggest_loguniform("y", 20, 30)
tests/test_dashboard.py:        y = trial.suggest_loguniform("y", 10, 20)
tests/trial_tests/test_fixed.py:    assert trial.suggest_loguniform("x", 0.1, 1.0) == 0.99
tests/trial_tests/test_fixed.py:        trial.suggest_loguniform("y", 0.0, 1.0)
tests/trial_tests/test_frozen.py:        b = trial.suggest_loguniform("b", 0.1, 10.0)
tests/trial_tests/test_frozen.py:    assert trial.suggest_loguniform("x", 0.1, 1.0) == 0.99
tests/trial_tests/test_frozen.py:        trial.suggest_loguniform("y", 0.0, 1.0)
tests/trial_tests/test_trial.py:    x4 = trial.suggest_loguniform("x2", 1e-5, 1e-3)
tests/trial_tests/test_trial.py:        trial.suggest_loguniform("x", 10, 20)
tests/trial_tests/test_trial.py:        trial.suggest_loguniform("x", 10, 20)
tests/trial_tests/test_trial.py:        trial.suggest_loguniform("x", 10, 30)
tests/trial_tests/test_trial.py:        assert trial.suggest_loguniform("b", 1.0, 1.0) == 1.0  # Suggesting a param.
tests/trial_tests/test_trial.py:        assert trial.suggest_loguniform("b", 1.0, 1.0) == 1.0  # Suggesting the same param.
tutorial/10_key_features/003_efficient_optimization_algorithms.py:    alpha = trial.suggest_loguniform("alpha", 1e-5, 1e-1)
tutorial/20_recipes/003_attributes.py:    svc_c = trial.suggest_loguniform("svc_c", 1e-10, 1e10)
```

</details>

**Comment by crcrpar** [comment]
Basically I want to replace *uniform with *float as possible entirely except for the implementations of *uniform methods.

Some calls of `trials.suggest_(float, loguniform, discreteuniform) are left there deliberately if the caller functions names include uniform. Otherwise, I overlooked them. Let me check them again.

**Comment by crcrpar** [comment]
I'd like this pull request to get "squash & merge" as this pull request has a bunch of meaningless commits.

**Comment by himkt** [comment]
I'm sorry if I misunderstood something but I'm wondering we can also update tutorials.

https://github.com/crcrpar/optuna/blob/code-fix/suggest-float/tutorial/20_recipes/005_user_defined_sampler.py#L20
https://github.com/crcrpar/optuna/blob/code-fix/suggest-float/tutorial/20_recipes/005_user_defined_sampler.py#L109-L110

It also needs to update an error massage, as well.
(e.g. `Only `suggest_float` with `log=False` is supported.`)

https://github.com/crcrpar/optuna/blob/code-fix/suggest-float/tutorial/20_recipes/005_user_defined_sampler.py#L78

**Comment by himkt** [comment]
> I'd like this pull request to get "squash & merge" as this pull request has a bunch of meaningless commits.

Gotcha, I'll merge this PR with squashing!

**Review by himkt — APPROVED** [review]
Thank you for the patience. LGTM.

**Pull request merged**